pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/PartitionUpsertMetadataManager.java

```diff
@@ -64,6 +64,11 @@ public interface PartitionUpsertMetadataManager extends Closeable {
    */
   void addSegment(ImmutableSegment segment);
 
+  /**
+   * Different from adding a segment, when preloading a segment, the upsert metadata may be updated more efficiently.
```

Comment:
nit: make this more informative by saying "upsert metadata will be directly updated instead of read-compare-update for each primary key"